### DISCUSSION TITLE
Include temperature sensors of solar thermal system

### DISCUSSION THREAD

**Discussion by ArneHi** [emsesp/EMS-ESP32#1526] [Q&A] [answered]
Hi Guys,

a couple of days ago I got the S3 Gateway from BBQKees, integrated it with Home Assistant and so far I love it. Only thing I hate is that I had such a hard time until I learned about EMS-ESP 😃

The facts about my heating are:
- Bosch/Junkers Cerapur ZSB 14-5
- Bosch CW400
- Bosch B-sol 100-2 Solar Controller for dhw heating

The solar Controller has a Display where you can see the temperature of the collectors and the buffer and some other basic information.

The spare part list regarding the sensors says something about 20k for the collector and NTC RD 6,0 for the buffer:
![Screenshot_20231222-225538](https://github.com/emsesp/EMS-ESP32/assets/102301271/d9f7283c-a637-43f8-88ee-76c92a3ee619)

And about the wiring of the solar controller:
![Screenshot_20231223-224410](https://github.com/emsesp/EMS-ESP32/assets/102301271/effb054d-da9f-43ba-bc0f-b8b1395b16b9)

So my simple question is: Is there any way to get the sensor data into the S3 Gateway? 😃
What I learned so far is that I cant just wire the cables together. The S3 doesn't find the sensor and somehow the solar controller also seems to get confused when both grounds are connected. 

Moreover I didn't find any approach to split or to replay the sensor data by googleing so far. For the buffer sensor probe or whatever I probably could use another seperat sensor but to add another one in the collector on the roof would just not be feasible for me.

I would be very happy If someone already solved that problem or if you guys could point me into the right direction since I'm having a hard time to learn all the good stuff about electronics.

Thank you very much,
Arne

**Comment by MichaelDvP** [comment, marked as answer]
This module does not have a ems connection, it's standalone. You can not read any value from it. 
You can:
- replace it by a ems-module, i think Bosch MS100 maybe compatible (maybe also older Junkers ISM1, check your manuals for compatibility).
-  use some dallas DS18B20 sensors on the ems-esp and connect them to the solar pipes and buffer. You will not have the collector temperature and no indication of pump running, but a little feedback what the solar do.
- use a camera and read the value from display, esp-cam with [this software](https://github.com/jomjol/AI-on-the-edge-device) is really cheap and nice ;-)

**Reply by ArneHi** [reply]
Thank you for your reply. I already feared that it wouldn't be able to connect.

I also thought about some sort of CV or OCR to collect values from the display but then I also need a fingerbot to cycle through the values and I feel that the solution might be a bit janky overall. 😃 
I think I will settle for now with another temperaure probe in the buffer if I'm able to fit and call it a day.

Even though I think that it wouldn't be worth for me to upgrade the controller for ~300€, could I wire it to the same S3 Gateway by wire while using the service jack on the other devices or would I need another gateway?

Thanks!

**Reply by MichaelDvP** [reply]
> I think I will settle for now with another temperaure probe in the buffer if I'm able to fit and call it a day.

You can add multiple dallas-sensors, see: https://bbqkees-electronics.nl/wiki/gateway/adding-external-sensors.html
This could look like:
![grafik](https://github.com/emsesp/EMS-ESP32/assets/59284019/bb130c0e-91ad-4cfb-9b69-c171c16c7a5f)


> Even though I think that it wouldn't be worth for me to upgrade the controller for ~300€, could I wire it to the same S3 Gateway by wire while using the service jack on the other devices or would I need another gateway?

The ems modules are connected to ems-bus and can be configured via CW400. You will have all values/parameters in your existing ems-esp. You can try to get a used one cheaper from ebay. 
This looks like this:
![grafik](https://github.com/emsesp/EMS-ESP32/assets/59284019/48c75590-d745-47d5-a739-fbf8b36e5c4e)
